### PR TITLE
task/WA-144: v3 ADCIRC app for DesignSafe

### PR DESCRIPTION
**Pull request by edmondsgarrett**
### Overview ###
Convert ADCIRC, PADCIRC, and PADCIRC SWAN DesignSafe apps to tapis v3

### Includes ###
[WA-144](https://tacc-main.atlassian.net/browse/WA-144)

### Changes ###
- Created app definitions, wrappers, and tapis job profiles (as needed) for ADCIRC, PADCIRC, and PADCIRC SWAN to run on Frontera

### Testing ###
- All three apps deployed to dev.cep.tacc.utexas.edu Applications -> DesignSafe
- Please use test cases in Community Data (`adcirc/adcirc_test` for ADCIRC, `padcirc/padcirc_test` for PADCIRC + PADCIRC SWAN)
  

### Notes ###
- Apps will eventually move to Stampede3, need to coordinate getting these apps installed there
- `archiveFilter` for these apps is not functioning correctly for >1 file in list, will modify apps as needed for any future fix. _Update 4-12-24: Fix up in tapis jobs dev + staging, `archiveFilter` excludes now functioning as expected._

### CODE REVIEW

File: applications/padcirc-swan-frontera/app.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                "autoMountLocal": true,
+                "sourceUrl": null,
+                "targetPath": "padcircSwan-inputs",
+                "envKey": "inputDirectory"
+            }
+        ],
+        "fileInputArrays": [],
+        "nodeCount": 3,
+        "coresPerNode": 56,
+        "memoryMB": 192,
```
Comment (rstijerina) [inline applications/padcirc-swan-frontera/app.json:77]:
```suggestion
        "memoryMB": 192000,
```

File: applications/padcirc-frontera/profile.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "padcirc",
```
Comment (rstijerina) [inline applications/padcirc-frontera/profile.json:2]: Can we add the version to the profile name? e.g. 55.01

File: applications/padcirc-frontera/profile.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "description": "Modules to load for PADCIRC",
+    "moduleLoads": [
+        {
+            "modulesToLoad": [
+                "intel/19.1.1",
+                "impi/19.0.9",
+                "netcdf/4.6.2",
+                "hdf5/1.10.4",
```
Comment (rstijerina) [inline applications/padcirc-frontera/profile.json:10]: Do you happen to know if only padcirc uses the hdf5 package?

File: applications/adcirc-frontera/app.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                "autoMountLocal": true,
+                "sourceUrl": null,
+                "targetPath": "adcirc-inputs",
+                "envKey": "inputDirectory"
+            }
+        ],
+        "fileInputArrays": [],
+        "nodeCount": 1,
+        "coresPerNode": 56,
+        "memoryMB": 192,
```
Comment (rstijerina) [inline applications/adcirc-frontera/app.json:77]:
```suggestion
        "memoryMB": 192000,
```

File: applications/adcirc-frontera/app.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
+{
+    "id": "adcirc",
```
Comment (rstijerina) [inline applications/adcirc-frontera/app.json:2]: Can you add the paths to the subdirs of each `app.json` to `initialize_tenant.py`? See figuregen for reference

Comment (rstijerina) [reply applications/adcirc-frontera/app.json:2]: https://github.com/TACC/WMA-Tapis-Templates/blob/561174af3bba776a26c99d28c234fcbbfcd9cff6/initialize_tenant.py#L157

File: applications/adcirc-frontera/app.json (hunk 4 above)
Comment (fnets) [reply applications/adcirc-frontera/app.json:77]: Don't forget to do this for the other app definitions, too.

File: applications/padcirc-frontera/profile.json (hunk 3 above)
Comment (edmondsgarrett) [reply applications/padcirc-frontera/profile.json:10]: Ah I didn't notice this differs for PADCIRC + PADCIRC SWAN in DS - I'll assume it is fine to add for PADCIRC SWAN as well.

Comment (edmondsgarrett) [reply applications/padcirc-frontera/profile.json:10]: I can't find any documentation about packages to load to run ADCIRC, and the only public app on DS for ADCIRC doesn't load hdf5.

**Review by rstijerina — APPROVED** [review]

File: applications/padcirc-frontera/app.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
+{
+    "id": "padcirc",
+    "version": "55.01",
+    "description": "PADCIRC is the parallel version of the ADCIRC which is optimized for enhanced performance on multiple computer nodes to run very large models. It includes MPI library calls to allow it to operate at high efficiency on parallel machines.",
+    "owner": "${apiUserId}",
+    "enabled": true,
+    "runtime": "ZIP",
+    "runtimeVersion": null,
+    "runtimeOptions": null,
+    "containerImage": "tapis://cloud.data/corral/tacc/aci/CEP/applications/v3/padcirc/v55/padcirc.zip",
```
Comment (fnets) [inline applications/padcirc-frontera/app.json:10]: I get `permission denied` for this file when I try to run the app published in dev.cep

File: applications/adcirc-frontera/app.json [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    "version": "55.01",
+    "description": "ADCIRC, ADvanced CIRCulation, is a system of computer programs for solving time-dependent, free surface circulation and transport problems in two and three dimensions. These programs utilize the finite element method in space allowing the use of highly flexible and unstructured grids.",
+    "owner": "${apiUserId}",
+    "enabled": true,
+    "runtime": "ZIP",
+    "runtimeVersion": null,
+    "runtimeOptions": null,
+    "containerImage": "tapis://cloud.data/corral/tacc/aci/CEP/applications/v3/adcirc/v55/adcirc.zip",
```
Comment (fnets) [inline applications/adcirc-frontera/app.json:10]: I get permission denied for this file when I try to run the app published in dev.cep

**Comment by edmondsgarrett** [comment]
> I'm only seeing ADCIRC in dev.cep. Did the other two get published? Or did they need a CMS entry in the apps tray or something?

Neglected to publish them to the staging tenant, all there now!

**Pull request merged**